imap_processing/ena_maps/ena_maps.py

```diff
@@ -867,6 +870,304 @@ def __init__(
+        self,
+        rect_spacing_deg: float,
+        value_keys: list[str],
+        max_subdivision_depth: int = MAX_SUBDIV_RECURSION_DEPTH,
+    ) -> tuple[RectangularSkyMap, dict[str, np.typing.NDArray]]:
+        """
+        Interpolate a healpix map to a rectangular map using recursive subdivision.
+
+        Parameters
+        ----------
+        rect_spacing_deg : float
+            The spacing of the rectangular map in degrees.
+        value_keys : list[str]
+            The names of the values to interpolate from the healpix map.
+            Each must be independently interpolated because the subdivision depth
+            depends on the gradient of the value between adjacent healpix pixels.
+        max_subdivision_depth : int, optional
+            The maximum depth of recursion for subdivision,
+            by default MAX_SUBDIV_RECURSION_DEPTH.
+
+        Returns
+        -------
+        tuple[RectangularSkyMap, dict[str, np.typing.NDArray]]
+            A RectangularSkyMap containing the interpolated values, and a dictionary of
+            each value and its corresponding subdivision depth by pixel.
+        """
+        # Begin by defining the rectangular map we want to create, which must be
+        # in the same spice reference frame as the healpix map
+        rect_map = RectangularSkyMap(
+            spacing_deg=rect_spacing_deg,
+            spice_frame=self.spice_reference_frame,
+        )
+
+        # Depending on the maximum recursion depth, the number of pixels in the
+        # RectangularSkyMap, and the number of value keys, and especially on the
+        # gradients of the values, the number of operations can be very large, so
+        # log key information about the expected number of operations.
+        approx_max_operations = (
+            (4**max_subdivision_depth) * self.num_points * len(value_keys)
+        )
+        logger.info(
+            f"Converting from a HealpixSkyMap(nside={self.nside}) to a "
+            f"RectangularSkyMap(spacing_deg={rect_spacing_deg}) with recursive "
+            "subdivision.\n The maximum recursion depth is "
+            f"{max_subdivision_depth}, yielding a maximum number of healpix calls"
+            f" of {approx_max_operations:.3e}."
+        )
+
+        # Dict to hold the subdivision depth by pixel for each value key
+        subdiv_depth_dict = {}
+        for value_key in value_keys:
+            self.data_1d[value_key]
+
+            # For each of the values, calculate each pixel's value with
+            # recursive subdivision. Unfortunately, this must be done independently
+            # for each value key.
+            healpix_values_array = self.data_1d[value_key]
+            best_value_and_recursion_depth_by_pixel = [
+                self.get_rect_pixel_value_recursive_subdivs(
+                    rect_pix_center_lon_lat=lon_lat,
+                    rect_pix_spacing_deg=rect_map.spacing_deg,
+                    value_key=value_key,
```

Comment:
Rather than value_key, can you pass the value array in? `self.data_1d[value_key]` I think. That way this could actually be a self-contained helper function and not rely on "self". I would rather project a value than a name, and this might also save many __getitem__ calls later too?

Reply:
Yeah, I think that is fine, a staticmethod then, or just a general helper somewhere if you prefer. I don't have a preference either way.

Reply:
Darn, actually, I remembered why these should actually receive the self key, because the lowest-level method: `calculate_rect_pixel_value_from_healpix_map_n_subdivisions` needs to get `self.nside, self.nested`. 

We could pass those in separately, but I think it's neater that they have access to  properties of the map which calls them. I'll replace value_key with value_array though still. Sound reasonable?